sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -750,6 +750,8 @@ func (host *nodeLanguageHost) execNodejs(ctx context.Context, req *pulumirpc.Run
 		// error messages in the engine.
 		contract.IgnoreError(os.Stdout.Sync())
 		contract.IgnoreError(os.Stderr.Sync())
+		// Close the write end of the pipe to signal to the sniffer that it should stop scanning.
+		contract.IgnoreError(w.Close())
```

Comment:
Without this we were waiting for the scan timeout if node crashed with a non-OOM error, or if we didn't detect the OOM.